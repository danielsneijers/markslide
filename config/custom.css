/**
 * This file will always be loaded after the theme is applied,
 * so if you have any custom styles or theme styles you want
 * to override, this is the place to be.
 *
 ***** IMAGES ******************************************************
 * If you want to use images from the config folder, use /images
 * as the base of you url, as they're server from dist/images in the
 * build
 */

.sassExample {
  background-color: #16232A;
}
