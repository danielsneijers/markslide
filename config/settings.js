module.exports = {
  title: 'Presenter',
  progressBar: true,
  theme: 'minimal',
  autoZoom: true,
  colors: {
    blue: '#56A9F6',
    red: '#EA5E5B'
  }
}
