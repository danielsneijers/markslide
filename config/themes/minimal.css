/* stylelint-disable selector-class-pattern, selector-max-compound-selectors */
html,
body {
  font-size: 1.5em;
  font-family: -apple-system, ".SFNSText-Regular", "San Francisco", "Roboto", "Segoe UI", "Helvetica Neue", "Lucida Grande", sans-serif; /* stylelint-disable-line max-line-length */
  line-height: 1.5em;
}

h1,
h2,
h3,
h4,
h5,
h6 {
  margin-bottom: 1rem;
  line-height: 1.2em;
}

h1 {
  font-size: 2.441em;
  font-weight: bold;
}

h2 {
  font-size: 1.953em;
}

h3 {
  font-size: 1.563em;
}

h4 {
  font-size: 1.25em;
}

p {
  margin-bottom: 1rem;
  line-height: 1.5em;
  font-size: 1.953em;
}

small {
  color: #A7AAA9;
  font-size: .7em;
}

/*
 * Highlighted code styles
 */
code,
pre {
  color: #C5C8C6;
  text-shadow: 0 1px rgba(0, 0, 0, .3);
  font-family: "Fira Code", Monaco, Consolas, 'Courier New', Courier, monospace;
  direction: ltr;
  text-align: left;
  white-space: pre-wrap;
  word-spacing: normal;
  word-break: normal;
  line-height: 1.5;
  tab-size: 4;
  hyphens: none;
}

/* Code blocks */
pre {
  padding: 0 .2em;
  margin: .5em 0;
  overflow: auto;
  border-radius: .3em;
}

:global .token.comment,
:global .token.prolog,
:global .token.doctype,
:global .token.cdata {
  color: #7C7C7C;
}

:global .token.punctuation {
  color: #C5C8C6;
}

.namespace {
  opacity: .7;
}

:global .token.property,
:global .token.keyword,
:global .token.tag {
  color: #96CBFE;
}

:global .token.class-name {
  color: #FFFFB6;
  text-decoration: underline;
}

:global .token.boolean,
:global .token.constant {
  color: #9C9;
}

:global .token.symbol,
:global .token.deleted {
  color: #F92672;
}

:global .token.number {
  color: #FF73FD;
}

:global .token.selector,
:global .token.attr-name,
:global .token.string,
:global .token.char,
:global .token.builtin,
:global .token.inserted {
  color: #A8FF60;
}

:global .token.variable {
  color: #C6C5FE;
}

:global .token.operator {
  color: #EDEDED;
}

:global .token.entity {
  color: #FFFFB6;
}

:global .token.url {
  color: #96CBFE;
}

:global .language-css .token.string,
:global .style .token.string {
  color: #87C38A;
}

:global .token.atrule,
:global .token.attr-value {
  color: #F9EE98;
}

:global .token.function {
  color: #DAD085;
}

:global .token.regex {
  color: #E9C062;
}

:global .token.important {
  color: #FD971F;
}

:global .token.important,
:global .token.bold {
  font-weight: bold;
}

:global .token.italic {
  font-style: italic;
}

:global .token.entity {
  cursor: help;
}
