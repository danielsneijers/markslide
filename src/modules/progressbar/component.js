// @flow
import React from 'react'
import CSS from './style.css'
import type { ProgressBarProps } from './types'

const ProgressBar = ({ offset }: ProgressBarProps): React$Element<any> => {
  const style = { transform: `translateX(-${offset}%)` }

  return <span className={CSS.progressBar} style={style} />
}

export default ProgressBar
