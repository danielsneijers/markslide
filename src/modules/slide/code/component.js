// @flow
import React from 'react'
import { compose } from 'ramda'
import classNames from 'classnames'
import { fullScreenOption, autoScaleContent } from 'higher-order-components'
import CSS from './style.css'

type Props = {
  index: number,
  content: string,
  meta: {
    class?: string // eslint-disable-line react/no-unused-prop-types
  },
  nextSlide: Function,
  style: {}
};

const CodeSlide = ({ content, index, meta, style, nextSlide }: Props) => {
  const slideClasses = classNames(
    CSS.slide,
    meta.class,
    'code-slide',
    `slide-${index}`
  )

  return (
    <div className={slideClasses} key={`slide-${index}`}>
      <div
        id={`slide-${index}`}
        key={Math.random()}
        className={CSS.code}
        style={{ opacity: 0, ...style }}
        onClick={nextSlide}
        dangerouslySetInnerHTML={{ __html: content }}
      />
    </div>
  )
}

const enhance = compose(fullScreenOption, autoScaleContent)

export default enhance(CodeSlide)
