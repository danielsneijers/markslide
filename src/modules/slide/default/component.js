// @flow
import React from 'react'
import { compose } from 'ramda'
import classNames from 'classnames'
import CSSTransitionGroup from 'react-transition-group/CSSTransitionGroup'
import { fullScreenOption, autoScaleContent } from 'higher-order-components'
import CSS from './style.css'

type Props = {
  index: number,
  content: string,
  meta: {
    class?: string // eslint-disable-line react/no-unused-prop-types
  },
  nextSlide: Function,
  style: {}
};

const Slide = ({ content, index, nextSlide, meta, style }: Props) => {
  const slideClasses = classNames(
    CSS.slide,
    meta.class,
    'regular-slide',
    `slide-${index}`
  )

  return (
    <CSSTransitionGroup
      transitionName='slide'
      transitionAppearTimeout={500}
      transitionEnterTimeout={500}
      transitionLeaveTimeout={300}
      transitionAppear
      component='div'
      className={slideClasses}
      key={`slide-${index}`}
      onClick={nextSlide}
    >
      <div
        id={`slide-${index}`}
        style={{ opacity: 0, ...style }}
        dangerouslySetInnerHTML={{ __html: content }}
      />
    </CSSTransitionGroup>
  )
}

const enhance = compose(fullScreenOption, autoScaleContent)

export default enhance(Slide)
