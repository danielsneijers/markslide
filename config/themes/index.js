import { theme } from '../settings'

/**
 * Function that dynamically imports the theme specified in settings.js
 * See https://webpack.js.org/guides/dependency-management/#require-context
 * for more details
 */
export function applyTheme () {
  const context = require.context('.', true, /\.css$/)

  context.keys().forEach(function (key) {
    if (key.includes(theme)) {
      context(key)
    }
  })

  // eslint-disable-next-line import/no-webpack-loader-syntax
  require('!style-loader!css-loader!postcss-loader!../custom.css')
}
