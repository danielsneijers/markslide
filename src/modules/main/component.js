// @flow
import React, { PureComponent } from 'react'
import { Switch, Route, Redirect } from 'react-router'
import { applyTheme } from 'config/themes'
import { keyboardNavigation } from 'higher-order-components'
import { ProgressBarContainer } from 'modules/progressbar'
import CSS from './style.css'

type Props = {
  renderRouteContainer: () => React$Element<any>
};

export class Main extends PureComponent {
  props: Props;

  componentWillMount () {
    applyTheme()
  }

  render (): React$Element<any> {
    const { renderRouteContainer } = this.props

    return (
      <div className={CSS.wrapper}>
        <ProgressBarContainer />
        <Switch>
          <Route path='/:slide' render={renderRouteContainer} />
          <Redirect exact from='/' to='/1' />
        </Switch>
      </div>
    )
  }
}

export default keyboardNavigation(Main)
