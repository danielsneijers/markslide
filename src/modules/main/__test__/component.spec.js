import React from 'react'
import { shallow } from 'enzyme'
import * as Themes from 'config/themes'
import { Main } from '../component'

let tree, instance
const routeContainer = <h1>Hello Test!</h1>

describe('components/Slide', () => {
  afterAll(() => {
    jest.resetAllMocks()
  })

  beforeEach(() => {
    Themes.applyTheme = jest.fn()
    tree = shallow(<Main routeContainer={routeContainer} />)
    instance = tree.instance()
  })

  it('renders correctly', () => {
    expect(tree).toMatchSnapshot()
  })

  it('applies a Theme on mount', () => {
    Themes.applyTheme.mockClear()

    expect(Themes.applyTheme).not.toHaveBeenCalled()

    instance.componentWillMount()

    expect(Themes.applyTheme).toHaveBeenCalledTimes(1)
  })
})
