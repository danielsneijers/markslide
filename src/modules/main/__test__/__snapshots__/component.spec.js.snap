// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main/component renders correctly 1`] = `
<div
  className="wrapper"
>
  <Connect(ProgressBar) />
  <Switch>
    <Route
      path="/:slide"
    />
    <Redirect
      exact={true}
      from="/"
      push={false}
      to="/1"
    />
  </Switch>
</div>
`;
